src/transformers/modeling_utils.py

```diff
@@ -2470,8 +2470,24 @@ def save_pretrained(
                 current_peft_config = self.peft_config[active_adapter]
                 current_peft_config.save_pretrained(save_directory)
 
+        # for offloaded modules
+        module_map = {}
+
         # Save the model
         if state_dict is None:
+            # if any model parameters are offloaded to the disk, make module map
+            if hasattr(self, "hf_device_map") and "disk" in self.hf_device_map.values():
```

Comment:
You forgot to include the case where "cpu" is in self.hf_device_map.values() also.  You can use the following check to see if we have offloading: `if isinstance(device_map, dict) and ("cpu" in device_map.values() or "disk" in device_map.values()):`